# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmCraneliftCompile.cpp
# Commit: 1ab8383d17ee
# Full Hash: 1ab8383d17ee1e5db164339236d64bcb7168b225
# Author: Lars T Hansen <lhansen@mozilla.com>
# Date: 2021-10-18 16:09:29
# Regressor Bug: 1727084
# File Overlap Count: 1
# Description:
#   Bug 1727084 - Memory64 - Atomic-rmw baseline operations. r=yury
#   
#   Atomic ADD, SUB, AND, OR, XOR, XCHG, CMPXCHG, as well as WAIT and
#   NOTIFY for the baseline compiler.
#   
# ==============================================================================

diff -r 6089b4360edc -r 1ab8383d17ee js/src/wasm/WasmCraneliftCompile.cpp
--- a/js/src/wasm/WasmCraneliftCompile.cpp	Mon Oct 18 07:31:55 2021 +0000
+++ b/js/src/wasm/WasmCraneliftCompile.cpp	Mon Oct 18 07:31:56 2021 +0000
@@ -99,11 +99,11 @@
     case BD_SymbolicAddress::PostBarrier:
       return SymbolicAddress::PostBarrierFiltering;
     case BD_SymbolicAddress::WaitI32:
-      return SymbolicAddress::WaitI32;
+      return SymbolicAddress::WaitI32M32;
     case BD_SymbolicAddress::WaitI64:
-      return SymbolicAddress::WaitI64;
+      return SymbolicAddress::WaitI64M32;
     case BD_SymbolicAddress::Wake:
-      return SymbolicAddress::Wake;
+      return SymbolicAddress::WakeM32;
     case BD_SymbolicAddress::Limit:
       break;
   }